### PR TITLE
LU-decomposition algorithm implementation

### PR DESCRIPTION
**Pull request by antonAce**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the comments
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms.Tests/Numeric/Decomposition/LUTests.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            Assert.AreEqual(expectedSolution, solution);
+        }
+        
+        [Test]
+        public void EliminateEquation_Case4X4()
+        {
+            // Arrange
+            var identityMatrix = new double[,] {{1.0, 2.0, -3.0, -1.0}, {0.0, -3.0, 2.0, 6.0}, {0.0, 5.0, -6.0, -2.0}, {0.0, -1.0, 8.0, 1.0}};
+            var coefficients = new double[,] {{0.0}, {-8.0}, {0.0}, {-8.0}};
+            var expectedSolution = new double[,] {{-1.0000000000000004}, {-2.0000000000000018}, {-1.0000000000000011}, {-2.0000000000000004}};
```
[the remark this answers is not in the document]
Comment (antonAce) [reply Algorithms.Tests/Numeric/Decomposition/LUTests.cs:106]: If I write this like {{-1}, {-2}, {-1}, {-2}}, the test will fail due to this error. I can implement check like |a-a*|<e for each member in solution vector. Ok?)

File: Algorithms/Numeric/Decomposition/LU.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        /// <summary>
+        /// Eliminates linear equations system represented as A*x=b, using LU-decomposition,
+        /// where A - matrix of equation coefficients, b - vector of absolute terms of equations.
+        /// </summary>
+        /// <param name="matrix">Matrix of equation coefficients.</param>
+        /// <param name="coefficients">Vector of absolute terms of equations.</param>
+        /// <returns>Vector-solution for linear equations system.</returns>
+        /// <exception cref="ArgumentException">Matrix of equation coefficients is not square shaped.</exception>
+        public static double[,] Eliminate(double[,] matrix, double[,] coefficients)
```
[the remark this answers is not in the document]
Comment (antonAce) [reply Algorithms/Numeric/Decomposition/LU.cs:75]: That's a correct remark, I'll fix it)

File: Algorithms.Tests/Numeric/Decomposition/LUTests.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        [Test]
+        public void FailOnDecomposeNonSquareMatrix()
+        {
+            // Arrange
+            var nonSquareMatrix = new double[,] {{1, 0, 0}, {0, 1, 0}, {0, 0, 1}, {0, 0, 0}};
+
+            // Act
+            void Act() => LU.Decompose(nonSquareMatrix ?? new double[,] {});
```
[the remark this answers is not in the document]
Comment (antonAce) [reply Algorithms.Tests/Numeric/Decomposition/LUTests.cs:64]: This is just a stub for fixing "Possible null pointer exception" error in stylecop. This is the only solution I came up with.

File: Algorithms.Tests/Numeric/Decomposition/LUTests.cs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            // Arrange
+            var identityMatrix = new double[,] {{1, 0, 0}, {0, 1, 0}, {0, 0, 1}};
+            var coefficients = new double[,] {{1}, {2}, {3}};
+            var expectedSolution = new double[,] {{1}, {2}, {3}};
+
+            // Act
+            var solution = LU.Eliminate(identityMatrix, coefficients);
+            
+            // Assert
+            Assert.AreEqual(expectedSolution, solution);
```
[the remark this answers is not in the document]
Comment (antonAce) [reply Algorithms.Tests/Numeric/Decomposition/LUTests.cs:82]: Yes)

File: Algorithms.Tests/Numeric/Decomposition/LUTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (siriak) [reply Algorithms.Tests/Numeric/Decomposition/LUTests.cs:64]: Ok, I'll take a look at it later

**Comment by antonAce** [comment]
@siriak, minor issues are fixed now! If you want to get familiar with this algorithm you can read about it in the book "Algorithms and Datastructures" by A. Levitin.

File: README.md [hunk 5]
```diff
@@ -21,6 +21,8 @@ This repository contains algorithms and data structures implemented in C# for ed
 		* [Naive solver](./Algorithms/Knapsack/NaiveKnapsackSolver.cs)
 		* [Dynamic Programming solver](./Algorithms/Knapsack/DynamicProgrammingKnapsackSolver.cs)
 	* [Numeric](./Algorithms/Numeric/)
+        * [Decomposition](./Algorithms/Numeric/Decomposition)
+            * [LU](./Algorithms/Numeric/Decomposition/LU.cs)
```
Comment (siriak) [inline README.md:25]:
```suggestion
		* [Decomposition](./Algorithms/Numeric/Decomposition)
			* [LU](./Algorithms/Numeric/Decomposition/LU.cs)
```

**Pull request merged**